frontend/public/components/modals/toleration-update-modal.tsx

```diff
@@ -0,0 +1,129 @@
+/* eslint-disable no-undef, no-unused-vars */
+
+import * as React from 'react';
+import { PromiseComponent, Dropdown } from '../utils';
+import {K8sKind, k8sPatch} from '../../module/k8s';
+
+import {createModalLauncher, ModalBody, ModalComponentProps, ModalSubmitFooter, ModalTitle} from '../factory';
+import * as _ from 'lodash-es';
+
+class TolerationUpdateModal extends PromiseComponent {
+  readonly state: TolerationUpdateModalState;
+
+  constructor(public props: TolerationUpdateModalProps) {
+    super(props);
+    this.state.tolerations = this.props.resource.spec.tolerations;
+  }
+
+  _submit = (e) => {
+    e.preventDefault();
+
+    // We just throw away any rows where the key is blank
+    const tolerations = _.reject(this.state.tolerations, t => _.isEmpty(t.key));
```

Comment:
This has a special meaning. We can't just throw away empty keys.

> An empty key with operator Exists matches all keys, values and effects which means this will tolerate everything.

https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/